src/Altinn.App.Core/Internal/LocaltestValidation.cs

```diff
@@ -0,0 +1,210 @@
+using System.Globalization;
+using System.Net;
+using System.Threading.Channels;
+using Altinn.App.Core.Configuration;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.App.Core.Internal;
+
+internal static class LocaltestValidationDI
+{
+    public static IServiceCollection AddLocaltestValidation(
+        this IServiceCollection services,
+        IConfiguration configuration
+    )
+    {
+        services.AddSingleton<LocaltestValidation>();
+        if (configuration.GetValue<bool>("GeneralSettings:DisableLocaltestValidation"))
+            return services;
+        services.AddSingleton<IHostedService>(provider => provider.GetRequiredService<LocaltestValidation>());
```

Comment:
Why doesn't this work?
```suggestion
        services.AddSingleton<IHostedService, LocaltestValidation>();
```

Reply:
Old habit of mine, but I think that becomes 2 different singletons since the service type is different?

Reply:
Ah, so this is a hack to be able to both have a generic `IHostedService` and have the `LocaltestValidation` as an injectable service.

Reply:
Which you need for testing in order to avoid `provider.GetRequiredServices<IHostedService>().OfType<LocaltestValidation>().Single()`?

Reply:
yup! which is not needed here at all (I started out doing things differently), so I'll cleanup